docs/BuildingASite.md

```diff
@@ -0,0 +1,65 @@
+# Building a Site Using Terra-dev-site
+Terra-dev-site dynamically builds a react-hash-routed site based on site configuration, navigation configuration and component configuration.
+
+## Getting Started
+- Install with [npmjs](https://www.npmjs.com/):
+    - `npm install terra-dev-site`
+
+## How It Works
+The site configuration file is the file loaded by the entry point during the webpack build. The site configuration provides the necessary navigation, component and application configuration to the site app such that the application can enable the application utilities and recursively build the menu navigation and content routes. The component configuration is truly the only configuration that must be provided, otherwise, Terra-dev-site provides the following default configurations:
+- `site.config.js`
+- `navigation.config.js`
+- `webpack.config.js`
+- `webpack.prod.config.js`
+
+## A Minimalist Approach
+To build a basic site that uses the default configuration, you must:
+
+1. Generate the component configuration. The easiest approach would be to add the `generate-config` script to the `package.json`:
+```
+"generate-config": "node node_modules/terra-dev-site/scripts/generate-component-config/generate-component-config.js",
+```
+This will save your configuration as `./generatedComponentConfig.js`.  **Note:** This script uses default search patterns which assumes a [specific file structure](https://github.com/cerner/terra-dev-site/blob/master/docs/ComponentConfig.md#generating-component-config).
+2. Create the site configuration file as `./site.config.js`.
```

Comment:
throw a line break above here or markup will just jam it into the previous paragraph